README.md
# Mini Media Player

[![](https://img.shields.io/github/release/kalkih/mini-media-player.svg?style=flat-square)](https://github.com/kalkih/mini-media-player/releases/latest)
[![](https://img.shields.io/travis/kalkih/mini-media-player.svg?style=flat-square)](https://travis-ci.org/kalkih/mini-media-player)

A minimalistic yet customizable media player card for [Home Assistant](https://github.com/home-assistant/home-assistant) Lovelace UI.

Inspired by [Custom UI: Mini media player](https://community.home-assistant.io/t/custom-ui-mini-media-player/40135) and [custom-lovelace](https://github.com/ciotlosm/custom-lovelace).

![Preview Image](https://user-images.githubusercontent.com/457678/47517460-9282d600-d888-11e8-9705-cf9ec3698c3c.png)


## Install

*This card is available in [HACS](https://github.com/custom-components/hacs) (Home Assistant Community Store)*

### Simple install

1. Download and copy `mini-media-player-bundle.js` from the [latest release](https://github.com/kalkih/mini-media-player/releases/latest) into your `config/www` directory.

2. Add a reference to `mini-media-player-bundle.js` inside your `configuration.yaml` or through the Home Assistant UI from the resource tab.

  ```yaml
  lovelace:
    resources:
      - url: /local/mini-media-player-bundle.js?v=1.13.0
        type: module
  ```

### CLI install

1. Move into your `config/www` directory

2. Grab `mini-media-player-bundle.js`

  ```console
  $ wget https://github.com/kalkih/mini-media-player/releases/download/v1.13.0/mini-media-player-bundle.js
  ```

3. Add a reference to `mini-media-player-bundle.js` inside your `configuration.yaml` or through the Home Assistant UI from the resource tab.

  ```yaml
  lovelace:
    resources:
      - url: /local/mini-media-player-bundle.js?v=1.13.0
        type: module
  ```

## Updating
1. Find your `mini-media-player-bundle.js` file in `config/www` or wherever you ended up storing it.

2. Replace the local file with the latest one attached in the [latest release](https://github.com/kalkih/mini-media-player/releases/latest).

3. Add the new version number to the end of the cards reference url in your `configuration.yaml` or through the Home Assistant.

  ```yaml
  lovelace:
    resources:
      - url: /local/mini-media-player-bundle.js?v=1.13.0
        type: module
  ```

*You may need to empty the browsers cache if you have problems loading the updated card.*

## Using the card

### Options

#### Card options
| Name | Type | Default | Since | Description |
|------|------|---------|-------|-------------|
| type | string | **required** | v0.1 | `custom:mini-media-player`
| entity | string | **required** | v0.1 | An entity_id from an entity within the `media_player` domain.
| name | string | optional | v0.6 | Override the entities friendly name.
| icon | string | optional | v0.1 | Specify a custom icon from any of the available mdi icons.
| tap_action | [action object](#action-object-options) | true | v0.7.0 | Action on click/tap.
| group | boolean | optional | v0.1 | Removes paddings, background color and box-shadow.
| hide | object | optional | v1.0.0 | Manage visible UI elements, see [hide object](#hide-object) for available options.
| artwork | string | default | v0.4 | `cover` to display current artwork in the card background, `full-cover` to display full artwork, `material` for alternate artwork display with dynamic colors, `none` to hide artwork, `full-cover-fit` for full cover without cropping.
| tts | object | optional | v1.0.0 | Show Text-To-Speech input, see [TTS object](#tts-object) for available options.
| source | string | optional | v0.7 | Change source select appearance, `icon` for just an icon, `full` for the full source name.
| sound_mode | string | optional | v1.1.2 | Change sound mode select appearance, `icon` for just an icon, `full` for the full sound mode name.
| info | string | optional | v1.0.0 | Change how the media information is displayed, `short` to limit media information to one row, `scroll` to scroll overflowing media info.
| volume_stateless | boolean | false | v0.6 | Swap out the volume slider for volume up & down buttons.
| volume_step | number | optional | v1.9.0 | Change the volume step size of the volume buttons and the volume slider (number between 1 - 100)<sup>[1](#option_foot1)</sup>.
| max_volume | number | optional | v0.8.2 | Specify the max vol limit of the volume slider (number between 1 - 100).
| min_volume | number | optional | v1.1.2 | Specify the min vol limit of the volume slider (number between 1 - 100).
| replace_mute | string | optional | v0.9.8 | Replace the mute button, available options are `play_pause` (previously `play`), `stop`, `play_stop`, `next`.
| toggle_power | boolean | true | v0.8.9 | Set to `false` to change the power button behaviour to `media_player.turn_on`/`media_player.turn_off`.
| idle_view | object | optional | v1.0.0 | Display a less cluttered view when idle, See [Idle object](#idle-object) for available options.
| background | string | optional | v0.8.6 | Background image, specify the image url `"/local/background-img.png"` e.g.
| speaker_group | object | optional | v1.0.0 | Speaker group management/multiroom, see [Speaker group object](#speaker-group-object) for available options.
| shortcuts | object | optional | v1.0.0 | Media shortcuts in a list or as buttons, see [Shortcut object](#shortcuts-object) for available options.
| scale | number | optional | v1.5.0 | UI scale modifier, default is `1`.

<a name="option_foot1"><sup>1</sup></a> Only supported on entities with `volume_level` attribute.

#### Idle object
| Name | Type | Default | Description |
|------|------|---------|-------------|
| when_idle | boolean | optional | Render the idle view when player state is `idle`.
| when_paused | boolean | optional | Render the idle view when player state is `paused`
| when_standby | boolean | optional | Render the idle view when player state is `standby`
| after | string | optional | Specify a number (minutes) after which the card renders as idle *(only supported on platforms exposing `media_position_updated_at`)*.

#### TTS object
| Name | Type | Default | Description |
|------|------|---------|-------------|
| platform | string | **required** | Specify [TTS platform](https://www.home-assistant.io/components/tts/), e.g. `google_translate` or `amazon_polly`, `cloud` for Nabu Casa, `alexa`<sup>[1](#tts_foot1)</sup> for ["Alexa as Media Player"](https://community.home-assistant.io/t/echo-devices-alexa-as-media-player-testers-needed/58639), `ga`<sup>[2](#tts_foot2)</sup><sup>[3](#tts_foot3)</sup> for use with [Google Assistant Webserver](https://community.home-assistant.io/t/community-hass-io-add-on-google-assistant-webserver-broadcast-messages-without-interrupting-music/37274) or [Assistant Relay](https://github.com/greghesp/assistant-relay), `sonos`<sup>[2](#tts_foot2)</sup> for use with modified [sonos_say script](https://github.com/kalkih/mini-media-player/issues/86#issuecomment-465541825), `webos`<sup>[4](#tts_foot4)</sup>.
| language | string | optional | The output language.
| entity_id | string/list | optional | The *entity_id* of the desired output entity or a list of *entity_id's*, can also be `all` to broadcast to all entities or `group` to target currently grouped speakers.
| volume | float | optional | Volume level of tts output (0 - 1), only supported by platform `sonos`.
| type | string | optional | `tts`, `announce` or `push`, defaults to `tts`, only supported by platform `alexa`, more info [here](https://github.com/keatontaylor/alexa_media_player/wiki/Notification-Component#functionality).

<a name="tts_foot1"><sup>1</sup></a> Does not support the `language` option.

<a name="tts_foot2"><sup>2</sup></a> Does not support `language` & `entity_id` options.

<a name="tts_foot3"><sup>3</sup></a> Requires a custom notify service named `ga_broadcast`, see example below.

<a name="tts_foot4"><sup>4</sup></a> Requires the card entity name to match the notify service name, if they don't match please specify the notify service name in the `entity_id` option.

```yaml
# configuration.yaml
notify:
  - name: ga_broadcast
    platform: rest
    resource: http://[xxx.x.x.xxx]:5000/broadcast_message
```

#### Speaker group object
See [Speaker group management](#speaker-group-management) for example usage.

**Supported platforms**
- sonos
- soundtouch
- squeezebox<sup>[2](#speaker_foot2)</sup>
- bluesound<sup>[2](#speaker_foot2)</sup>
- snapcast<sup>[2](#speaker_foot2)</sup>
- musiccast_yamaha<sup>[1](#speaker_foot1)</sup>
- linkplay<sup>[3](#speaker_foot3)</sup>
- media_player<sup>[4](#speaker_foot4)</sup>

| Name | Type | Default | Description |
|------|------|---------|-------------|
| entities | list | **required** | A list containing [speaker entities](#speaker-entity-object) of one of supported platforms, to enable group management of those speakers.
| platform | string | 'sonos' | Any supported multiroom platform e.g. `sonos`, `soundtouch`, `bluesound`, see **supported platforms** above.
| sync_volume | boolean | optional | Keep volume Synchronize between grouped speakers.
| expanded | boolean | optional | Make the speaker group list expanded by default.
| show_group_count | boolean | true | Have the number of grouped speakers displayed (if any) in the card name.
| icon | string | optional | Override default group button icon *(any mdi icon)*.
| group_mgmt_entity | string | optional | Override the player entity for the group management (Useful if you use a universal media_player as your entity but still want to use the grouping feature)
| supports_master | boolean | optional | Set to false if your multiroom system does not define one of the media players as master. Defaults to `true` and has not to be set to false for `squeezebox` for backward compatibility.

<a name="speaker_foot1"><sup>1</sup></a> Requires [custom component](https://github.com/ppanagiotis/pymusiccast), available in HACS.

<a name="speaker_foot2"><sup>2</sup></a> All features are not yet supported.

<a name="speaker_foot3"><sup>3</sup></a> Requires [custom component](https://github.com/nagyrobi/home-assistant-custom-components-linkplay#multiroom) for sound devices based on Linkplay chipset, available in HACS.

<a name="speaker_foot4"><sup>4</sup></a> HomeAssistant added join/unjoin services to the media_player. Future official integrations will implement these services (which are slightly different from the ones, which are already supported by this card) instead of implementing them in their own domain.

#### Speaker entity object
| Name | Type | Default | Description |
|------|------|---------|-------------|
| entity_id | string | **required** | The *entity_id* for the speaker entity.
| name | string | **required** | A display name.
| volume_offset | number | optional | Volume offset *(0-100)* when used with `sync_volume`.

#### Shortcuts object
See [card with media shortcuts](#card-with-media-shortcuts) for example usage.

| Name | Type | Default | Description |
|------|------|---------|-------------|
| list | list | optional | A list of [shortcut items](#shortcut-item-object) to be presented as a list, see shortcut item object.
| buttons | list | optional | A list of [shortcut items](#shortcut-item-object) to be presented as buttons.
| hide_when_off | boolean | false | Hide the shortcuts while the entity is off.
| columns | integer (1-6) | 2 | Specify the max number of buttons per row.
| column_height | number | optional | Specify the column height in pixels.
| label | string | `shortcuts...` | Specify a custom default label for the shortcut dropdown.
| attribute | string | optional | Specify any attribute exposed by the media player entity. The attribute value (if exists) is compared with shortcut `id`'s to distinguish selected/active shortcut<sup>[1](#shortcuts_foot1)</sup>.
| align_text | string | optional | Specify alignment of button icon/text `left`, `right`, `center`.

<a name="shortcuts_foot1"><sup>1</sup></a> Examples, `source` for active source or `sound_mode` for active sound mode.

#### Shortcut item object
| Name | Type | Default | Description |
|------|------|---------|-------------|
| name | string | optional | A display name.
| icon | string | optional | A display icon *(any mdi icon)*.
| image | string | optional | A display image.
| cover | string | optional | A cover image (only supported for button shortcuts).
| type | string | **required** | Type of shortcut. A media type: `music`, `tvshow`, `video`, `episode`, `channel`, `playlist` e.g. or an action type: `source`, `sound_mode`, `script` or `service`.
| id | string | **required** | The media identifier. The format of this is component dependent. For example, you can provide URLs to Sonos & Cast but only a playlist ID to iTunes & Spotify. A source/(sound mode) name can also be specified to change source/(sound mode), use together with type `source`/`sound_mode`. If type `script` specify the script name here or `service` specify the `<domain>.<service>`.
| data | list | optional | Extra service payload<sup>[1](#shortcut_foot1)</sup>.

<a name="shortcut_foot1"><sup>1</sup></a> Only compatible with `script` & `service` shortcuts, useful for sending variables to script.

#### Action object options
| Name | Type | Default | Options | Description |
|------|:----:|:-------:|:-----------:|-------------|
| action | string | `more-info` | `more-info` / `navigate` / `call-service`  / `url` / `fire-dom-event` / `none` | Action to perform.
| entity | string |  | Any entity id | Override default entity of `more-info`, when  `action` is defined as `more-info`.
| service | string |  | Any service | Service to call (e.g. `media_player.toggle`) when `action` is defined as `call-service`.
| service_data | object |  | Any service data | Service data to include with the service call (e.g. `entity_id: media_player.office`).
| navigation_path | string |  | Any path | Path to navigate to (e.g. `/lovelace/0/`) when `action` is defined as `navigate`.
| url | string |  | Any URL | URL to open when `action` is defined as `url`.
| new_tab | boolean | `false` | `true` / `false` | Open URL in new tab when `action` is defined as `url`.

#### Hide object
| Name | Type | Default | Description |
|------|------|---------|-------------|
| name | boolean | false | The name.
| icon | boolean | false | The entity icon.
| info | boolean | false | The media information.
| power | boolean | false | The power button.
| source | boolean | false | The source select.
| sound_mode | boolean | true | The sound_mode select.
| controls | boolean | false | The media playback controls.
| prev | boolean | false | The "previous" playback control button.
| next | boolean | false | The "next" playback control button.
| play_pause | boolean | false | The play/pause button in media playback controls.
| play_stop | boolean | true | The play/stop button in media playback controls.
| volume | boolean | false | The volume controls.
| volume_level | boolean | true | The current volume level in percentage.
| mute | boolean | false | The mute button.
| progress | boolean | false | The progress bar.
| runtime | boolean | true | The media runtime indicators.
| artwork_border | boolean | true | The border of the `default` artwork picture.
| power_state | boolean | true | Dynamic color of the power button to indicate on/off.
| icon_state | boolean | true | Dynamic color of the entity icon to indicate entity state.
| shuffle | boolean | true | The shuffle button (only for players with `shuffle_set` support).
| state_label | boolean | false | State labels such as `Unavailable` & `Idle`.


### Theme variables
The following variables are available and can be set in your theme to change the appearence of the card.
Can be specified by color name, hexadecimal, rgb, rgba, hsl, hsla, basically anything supported by CSS.

| name | Default | Description |
|------|---------|-------------|
| mini-media-player-base-color | var(--primary-text-color) & var(--paper-item-icon-color) | The color of base text & buttons
| mini-media-player-accent-color | var(--accent-color) | The accent color of UI elements
| mini-media-player-icon-color | --mini-media-player-base-color, var(--paper-item-icon-color, #44739e) | The color for icons
| mini-media-player-button-color | rgba(255,255,255,0.25) | The background color of shortcut and group buttons.
| mini-media-player-overlay-color | rgba(0,0,0,0.5) | The color of the background overlay
| mini-media-player-overlay-color-stop | 25% | The gradient stop of the background overlay
| mini-media-player-overlay-base-color | white | The color of base text, icons & buttons while artwork cover is present
| mini-media-player-overlay-accent-color | white | The accent color of UI elements while artwork cover is present

changelog.md
## v1.13.0
### NEW
- Support for the new default media player grouping functionality (#525) #519 #528 @micha91   
  This should allow support for all future multi-room integrations following the new standard
- Added new speaker group configuration `support_master` (#525) @micha91

### FIXED
- Issue where player progress continued when paused (#521) #346 @BoOmka

### TRANSLATIONS
-  Add russian translations (#510)

### DOCUMENTATION
- Clarify which section of configuration.yaml to edit (#526) @ronnix

## v1.12.1
### NEW
- Support for `fire-dom-event` #484 (#485) @RomRider

### FIXED
- Fixed missing speaker group checkboxes #418 (#464) @mcguiresean @RomRider
- Changed name of multiroom integration yamaha_musiccast -> musiccast_yamaha (#476) @winterscar
- Remove idle_view option from UI editor #468 (#502)
- Fix missing player icon #467 (#503)

### TRANSLATIONS
-  Add German translations #482 (#487) @Ruud2702


## v1.12.0
### NEW
- Added card to the Lovelace card picker (#436) @dcs8
- Added card editor UI (#442) @dcs8
- Added option to hide state labels
- Added option to override entity for group management (#458) @RomRider
- Hide next and prev buttons by default on unsupported entities (#433) @dmamontov

### FIXED
- Fixed missing speaker group checkboxes #418 (#464) @mcguiresean @RomRider
- Broken scale option #415
- Fix artwork fallback (#449) @CarsonF

### TRANSLATIONS
- Added Norwegian translations (#411) @kristomi
- Added CZ translation (#434) @KubaDee

### DOCUMENTATION
- Add mention for Nabu Casa, which seems to work (#456) @mrand
- Clarify docs regarding spotify playlist shortcut #441 (#461) @BenFR23


## v1.11.0
### NEW
- Ability to display volume level percentage #289 (#406)
- New tap action option, open url in new tab #227 (#405)

### FIXED
- Fix disappearing background #393 (#402)

### TRANSLATIONS
- Added Ukrainian translations (#400) @denysdovhan
- Improved Hungarian translations (#396) @nagyrobi


## v1.10.2
### NEW
- Add support for slider volume steps #365 (#382)

### FIXED
- Prevent jumping artwork when transitioning artwork material #378 (#381)
- Invisible default slider container color #387 (#390)
- Artwork not clearing post playback #376 (#391)
- Artwork url fallback if failing fetch of base64 image #385 (#392)

### CHANGES
- Minor changes to background & artwork transitions

### TRANSLATIONS
- Add Hungarian translations (#377) @nagyrobi
- Update tts label string (#384) @agneevX
- Translation support for master string (#380)


## v1.10.1
### NEW
- Add Polish translations (#370) @cyrosy
- Add French translations (#371) @bieniu

### FIXED
- Fixed invalid text & icon colors when using artwork cover option (#372)
- Fixed missing icon & broken default artwork #369 (#375)


## v1.10.0
### NEW
- Smoothly fade in & crossfade between cover artwork (#339)
- Add support for custom card specific translations #345 (#363)
- New cover artwork display option (material) #333 (#331) @SBado w inspo from @zsarnett

### FIXED
- Misaligned dropdown icon #325 (#332)
- Always expand dropdowns downwards #267 (#323)
- Fetch artwork completely before displaying to prevent flickering #321 (#339)
- Replace deprecated background theme variable #359 (#362) @agneevX
- Fix progress bar overflow in Safari when round card corners #352 (#364)

### DOCUMENTATION
- Added Yamaha musiccast multiroom information (#353) @ppanagiotis
- Added Linkplay multiroom platform information (#356) @nagyrobi

🌐 Feel free to submit translations by adding strings in the [translations](https://github.com/kalkih/mini-media-player/blob/dev/src/translations.js) file & submitting the changes in a PR.


## v1.9.1
### FIXED
- Fix broken shortcut column height #318 (#319)


## v1.9.0
### NEW
- Squeezebox multiroom support (#312) **(BETA)**
- Volume step size option for volume buttons #219 (#308) @craiq
- Added `cover` option for shortcut items #236 (#314)

### FIXED
- Fixed misaligned icons #305 (#310)
- Fixed misaligned multiroom button #305 (#313)
- Updated bluesound service calls for speaker group management #296 (#315)


## v1.8.1
### FIXED
- Changed icon element for HA > 0.110.0 support #302 (#304)


## v1.8.0
### ADDED
- Options to hide previous and/or next playback controls #233 (#275)
- Sync mute status on all speakers when `sync_volume` is enabled #262 (#274)
- New `tts` option `entity_id` parameter to target currently grouped speakers #268 (#273)

### FIXED
- Resume progress tracking correcty after switching between tabs #269 (#272)
- Fixed tts text color when artwork is present #270 (#271)
- Changed button element for HA > 0.110.0 support #297 (#301)

### DOCS
- Removed deprecated custom_updater instructions (#290) @inverse


## v1.7.0
### ADDED
- Support for Soundtouch multiroom (#165)

### FIXED
- Removed usage of deprecated thumbnail/artwork api endpoint #244 (#253)
- Fixed button theme variable not sticking to speaker group buttons


## v1.6.0
### ADDED
- Allow master speaker to leave speaker group #213
- New theme variable for raised buttons #212
- New theme variable for name font weight

### CHANGED
- Increased font weight of buttons

### FIXED
- Broken leave button for group management
- Disabled buttons are now properly disabled


## v1.5.2
### ADDED
- New icon state option, change entity icon color depending on state, disabled by default, see `icon_state` in `hide` option #186 #144

### CHANGED
- Reduced card paddings when using group option #210

### FIXED
- Broken dropdown menus in latest iOS #207

## v1.5.1
### ADDED
- Dynamic alignment of dropdown lists (#201)

### FIXED
- Broken dropdown lists (#201)
- Invisible icons (maybe?)

## v1.5.0
### ADDED
- Option to display image in shortcut #146 (#192) - (@tcarlsen)
- Tap action option #174 (#191) - (@darkkatarsis)
- Option for card UI scale #156 (#187)
- Theme variable for card UI scale #156 (#187)
- Theme variable for card background opacity #189
- Shortcut example with service call (#196) - (@exetico)

### FIXED
- Source dropdown not updating with backend #184

src/style.js
import { css } from 'lit-element';

const style = css`
  :host {
    overflow: visible !important;
    display: block;
    --mmp-scale: var(--mini-media-player-scale, 1);
    --mmp-unit: calc(var(--mmp-scale) * 40px);
    --mmp-name-font-weight: var(--mini-media-player-name-font-weight, 400);
    --mmp-accent-color: var(--mini-media-player-accent-color, var(--accent-color, #f39c12));
    --mmp-base-color: var(--mini-media-player-base-color, var(--primary-text-color, #000));
    --mmp-overlay-color: var(--mini-media-player-overlay-color, rgba(0,0,0,0.5));
    --mmp-overlay-color-stop: var(--mini-media-player-overlay-color-stop, 25%);
    --mmp-overlay-base-color: var(--mini-media-player-overlay-base-color, #fff);
    --mmp-overlay-accent-color: var(--mini-media-player-overlay-accent-color, --mmp-accent-color);
    --mmp-text-color: var(--mini-media-player-base-color, var(--primary-text-color, #000));
    --mmp-media-cover-info-color: var(--mini-media-player-media-cover-info-color, --mmp-text-color);
    --mmp-text-color-inverted: var(--disabled-text-color);
    --mmp-active-color: var(--mmp-accent-color);
    --mmp-button-color: var(--mini-media-player-button-color, rgba(255,255,255,0.25));
    --mmp-icon-color:
      var(--mini-media-player-icon-color,
        var(--mini-media-player-base-color,
          var(--paper-item-icon-color, #44739e)));
    --mmp-icon-active-color: var(--paper-item-icon-active-color, --mmp-active-color);
    --mmp-info-opacity: 0.75;
    --mmp-bg-opacity: var(--mini-media-player-background-opacity, 1);
    --mmp-artwork-opacity: var(--mini-media-player-artwork-opacity, 1);
    --mmp-progress-height: var(--mini-media-player-progress-height, 6px);
    --mdc-theme-primary: var(--mmp-text-color);
    --mdc-theme-on-primary: var(--mmp-text-color);
    --paper-checkbox-unchecked-color: var(--mmp-text-color);
    --paper-checkbox-label-color: var(--mmp-text-color);
    color: var(--mmp-text-color);
  }
  ha-card.--bg {
    --mmp-info-opacity: .75;
  }
  ha-card.--has-artwork[artwork='material'],
  ha-card.--has-artwork[artwork*='cover'] {
    --mmp-accent-color: var(--mini-media-player-overlay-accent-color, var(--mini-media-player-accent-color, var(--accent-color, #f39c12)));
    --mmp-text-color: var(--mmp-overlay-base-color);
    --mmp-text-color-inverted: #000;
    --mmp-active-color: rgba(255,255,255,.5);
    --mmp-icon-color: var(--mmp-text-color);
    --mmp-icon-active-color: var(--mmp-text-color);
    --mmp-info-opacity: 0.75;
    --paper-slider-container-color: var(--mini-media-player-overlay-color, rgba(255,255,255,.75)) !important;
    --mdc-theme-primary: var(--mmp-text-color);
    --mdc-theme-on-primary: var(--mmp-text-color);
    --paper-checkbox-unchecked-color: var(--mmp-text-color);
    --paper-checkbox-label-color: var(--mmp-text-color);
    --switch-checked-color: var(--mmp-accent-color);
    --switch-checked-button-color: var(--mmp-accent-color);
    --switch-checked-track-color: var(--mmp-accent-color);
    --switch-unchecked-color: var(--mmp-text-color);
    --switch-unchecked-button-color: var(--mmp-text-color);
    --switch-unchecked-track-color: var(--mmp-text-color);
    color: var(--mmp-text-color);
  }
  ha-card {
    cursor: default;
    display: flex;
    background: transparent;
    overflow: visible;
    padding: 0;
    position: relative;
    color: inherit;
    font-size: calc(var(--mmp-unit) * 0.35);
    --mdc-icon-button-size: calc(var(--mmp-unit));
    --mdc-icon-size: calc(var(--mmp-unit) * 0.6);
  }
  ha-card.--group {
    box-shadow: none;
    --mmp-progress-height: var(--mini-media-player-progress-height, 4px);
  }
  ha-card.--more-info {
    cursor: pointer;
  }
  .mmp__bg, .mmp-player, .mmp__container {
    border-radius: var(--ha-card-border-radius, 0);
  }
  .mmp__container {
    overflow: hidden;
    height: 100%;
    width: 100%;
    position: absolute;
    pointer-events: none;
    -webkit-transform: translateZ(0);
    transform: translateZ(0);
  }
  ha-card:before {
    content: '';
    padding-top: 0px;
    transition: padding-top .5s cubic-bezier(.21,.61,.35,1);
    will-change: padding-top;
  }
  ha-card.--initial .entity__artwork,
  ha-card.--initial .entity__icon {
    animation-duration: .001s;
  }
  ha-card.--initial:before,
  ha-card.--initial .mmp-player {
    transition: none;
  }
  header {
    display: none;
  }
  ha-card[artwork='full-cover'].--has-artwork:before {
    padding-top: 56%;
  }
  ha-card[artwork='full-cover'].--has-artwork[content='music']:before,
  ha-card[artwork='full-cover-fit'].--has-artwork:before {
    padding-top: 100%;
  }
  .mmp__bg {
    background: var(--ha-card-background, var(--card-background-color, var(--paper-card-background-color, white)));
    position: absolute;
    top: 0; right: 0; bottom: 0; left: 0;
    overflow: hidden;
    -webkit-transform: translateZ(0);
    transform: translateZ(0);
    opacity: var(--mmp-bg-opacity);
  }
  ha-card[artwork='material'].--has-artwork .mmp__bg,
  ha-card[artwork*='cover'].--has-artwork .mmp__bg {
    opacity: var(--mmp-artwork-opacity);
    background: transparent;
  }
  ha-card[artwork='material'].--has-artwork .cover {
    height: 100%;
    right: 0;
    left: unset;
    animation: fade-in 4s cubic-bezier(.21,.61,.35,1) !important;
  }
  ha-card[artwork='material'].--has-artwork .cover.--prev {
    animation: fade-in 1s linear reverse forwards !important;
  }
  ha-card[artwork='material'].--has-artwork .cover-gradient {
    position: absolute;
    height: 100%;
    right: 0;
    left: 0;
    opacity: 1;
  }
  ha-card.--group .mmp__bg {
    background: transparent;
  }
  ha-card.--inactive .cover {
    opacity: 0;
  }
  ha-card.--inactive .cover.--bg {
    opacity: 1;
  }
  .cover-gradient {
    transition: opacity .45s linear;
    opacity: 0;
  }
  .cover,
  .cover:before {
    display: block;
    opacity: 0;
    position: absolute;
    top: 0; right: 0; bottom: 0; left: 0;
    transition: opacity .75s linear, width .05s cubic-bezier(.21,.61,.35,1);
    will-change: opacity;
  }
  .cover:before {
    content: '';
    background: var(--mmp-overlay-color);
  }
  .cover {
    animation: fade-in .5s cubic-bezier(.21,.61,.35,1);
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center center;
    border-radius: var(--ha-card-border-radius, 0);
    overflow: hidden;
  }
  .cover.--prev {
    animation: fade-in .5s linear reverse forwards;
  }
  .cover.--bg {
    opacity: 1;
  }
  ha-card[artwork*='full-cover'].--has-artwork .mmp-player {
    background: linear-gradient(to top, var(--mmp-overlay-color) var(--mmp-overlay-color-stop), transparent 100%);
    border-bottom-left-radius: var(--ha-card-border-radius, 0);
    border-bottom-right-radius: var(--ha-card-border-radius, 0);
  }
  ha-card.--has-artwork .cover,
  ha-card.--has-artwork[artwork='cover'] .cover:before {
    opacity: .999;
  }
  ha-card[artwork='default'] .cover {
    display: none;
  }
  ha-card.--bg .cover {
    display: block;
  }
  ha-card[artwork='material'].--has-artwork .cover {
    background-size: cover;
  }
  ha-card[artwork='full-cover-fit'].--has-artwork .cover {
    background-color: black;
    background-size: contain;
  }
  .mmp-player {
    align-self: flex-end;
    box-sizing: border-box;
    position: relative;
    padding: 16px;
    transition: padding .25s ease-out;
    width: 100%;
    will-change: padding;
  }
  ha-card.--group .mmp-player {
    padding: 2px 0;
  }
  .flex {
    display: flex;
    display: -ms-flexbox;
    display: -webkit-flex;
    flex-direction: row;
  }
  .mmp-player__core {
    position: relative;
  }
  .entity__info {
    justify-content: center;
    display: flex;
    flex-direction: column;
    margin-left: 8px;
    position: relative;
    overflow: hidden;
    user-select: none;
  }
  ha-card.--rtl .entity__info {
    margin-left: auto;
    margin-right: calc(var(--mmp-unit) / 5);
  }
  ha-card[content='movie'] .attr__media_season,
  ha-card[content='movie'] .attr__media_episode {
    display: none;
  }
  .entity__icon {
    color: var(--mmp-icon-color);
  }
  .entity__icon[color] {
    color: var(--mmp-icon-active-color);
  }
  .entity__artwork, .entity__icon {
    animation: fade-in .25s ease-out;
    background-position: center center;
    background-repeat: no-repeat;
    background-size: cover;
    border-radius: 100%;
    height: var(--mmp-unit);
    width: var(--mmp-unit);
    min-width: var(--mmp-unit);
    line-height: var(--mmp-unit);
    margin-right: calc(var(--mmp-unit) / 5);
    position: relative;
    text-align: center;
    will-change: border-color;
    transition: border-color .25s ease-out;
  }
  ha-card.--rtl .entity__artwork,
  ha-card.--rtl .entity__icon {
    margin-right: auto;
  }
  .entity__artwork[border] {
    border: 2px solid var(--primary-text-color);
    box-sizing: border-box;
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
  }
  .entity__artwork[border][state='playing'] {
    border-color: var(--mmp-accent-color);
  }
  .entity__info__name,
  .entity__info__media[short] {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
  .entity__info__name {
    line-height: calc(var(--mmp-unit) / 2);
    color: var(--mmp-text-color);
    font-weight: var(--mmp-name-font-weight);
  }
  .entity__info__media {
    color: var(--secondary-text-color);
    max-height: 6em;
    word-break: break-word;
    opacity: var(--mmp-info-opacity);
    transition: color .5s;
  }
  .entity__info__media[short] {
    max-height: calc(var(--mmp-unit) / 2);
    overflow: hidden;
  }
  .attr__app_name {
    display: none;
  }
  .attr__app_name:first-child,
  .attr__app_name:first-of-type {
    display: inline;
  }
  .mmp-player__core[inactive] .entity__info__media {
    color: var(--mmp-text-color);
    max-width: 200px;
    opacity: .5;
  }
  .entity__info__media[short-scroll] {
    max-height: calc(var(--mmp-unit) / 2);
    white-space: nowrap;
  }
  .entity__info__media[scroll] > span {
    visibility: hidden;
  }
  .entity__info__media[scroll] > div {
    animation: move linear infinite;
  }
  .entity__info__media[scroll] .marquee {
    animation: slide linear infinite;
  }
  .entity__info__media[scroll] .marquee,
  .entity__info__media[scroll] > div {
    animation-duration: inherit;
    visibility: visible;
  }
  .entity__info__media[scroll] {
    animation-duration: 10s;
    mask-image: linear-gradient(to right, transparent 0%, black 5%, black 95%, transparent 100%);
    -webkit-mask-image: linear-gradient(to right, transparent 0%, black 5%, black 95%, transparent 100%);
  }
  .marquee {
    visibility: hidden;
    position: absolute;
    white-space: nowrap;
  }
  ha-card[artwork*='cover'].--has-artwork .entity__info__media,
  ha-card.--bg .entity__info__media {
    color: var(--mmp-media-cover-info-color);
  }
  .entity__info__media span:before {
    content: ' - ';
  }
  .entity__info__media span:first-of-type:before {
    content: '';
  }
  .entity__info__media span:empty {
    display: none;
  }
  .mmp-player__adds {
    margin-left: calc(var(--mmp-unit) * 1.2);
    position: relative;
  }
  ha-card.--rtl .mmp-player__adds {
    margin-left: auto;
    margin-right: calc(var(--mmp-unit) * 1.2);
  }
  .mmp-player__adds > *:nth-child(2) {
    margin-top: 0px;
  }
  mmp-powerstrip {
    flex: 1;
    justify-content: flex-end;
    margin-right: 0;
    margin-left: auto;
    width: auto;
    max-width: 100%;
  }
  mmp-media-controls {
    flex-wrap: wrap;
  }
  ha-card.--flow mmp-powerstrip {
    justify-content: space-between;
    margin-left: auto;
  }
  ha-card.--flow.--rtl mmp-powerstrip {
    margin-right: auto;
  }
  ha-card.--flow .entity__info {
    display: none;
  }
  ha-card.--responsive .mmp-player__adds {
    margin-left: 0;
  }
  ha-card.--responsive.--rtl .mmp-player__adds {
    margin-right: 0;
  }
  ha-card.--responsive .mmp-player__adds > mmp-media-controls {
    padding: 0;
  }
  ha-card.--progress .mmp-player {
    padding-bottom: calc(16px + calc(var(--mini-media-player-progress-height, 6px) - 6px));
  }
  ha-card.--progress.--group .mmp-player {
    padding-bottom: calc(10px + calc(var(--mini-media-player-progress-height, 6px) - 6px));
  }
  ha-card.--runtime .mmp-player {
    padding-bottom: calc(16px + 16px + var(--mini-media-player-progress-height, 0px));
  }
  ha-card.--runtime.--group .mmp-player {
    padding-bottom: calc(16px + 12px + var(--mini-media-player-progress-height, 0px));
  }
  ha-card.--inactive .mmp-player {
    padding: 16px;
  }
  ha-card.--inactive.--group .mmp-player {
    padding: 2px 0;
  }
  .mmp-player div:empty {
    display: none;
  }
  @keyframes slide {
    100% { transform: translateX(-100%); }
  }
  @keyframes move {
    from { transform: translateX(100%); }
    to { transform: translateX(0); }
  }
  @keyframes fade-in {
    from { opacity: 0; }
    to { opacity: 1; }
  }
  ha-switch {
    padding: 16px 6px;
  }
  .side-by-side {
    display: flex;
  }
  .side-by-side > * {
    flex: 1;
    padding-right: 4px;
  }
`;

export default style;
